src/anomalib/models/components/base/memory_bank_module.py

```diff
@@ -0,0 +1,38 @@
+"""Memory Bank Module."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+from abc import abstractmethod
+
+import torch
+from torch import nn
+
+
+class MemoryBankMixin(nn.Module):
```

Comment:
I think the reason this is not working that we probably need to inherit from `LightningModule` instead of `nn.Module`. But in that case I am not sure if this approach might work as we might have to implement configure_optimizers, and all those necessary methods.